tsdb/record/record.go

```diff
@@ -304,6 +304,10 @@ func (d *Decoder) Samples(rec []byte, samples []RefSample) ([]RefSample, error)
 		baseRef  = dec.Be64()
 		baseTime = dec.Be64int64()
 	)
+	// Allow 1 byte for each varint and 8 for the value; the output slice must be at least that big.
+	if minSize := dec.Len() / (1 + 1 + 8); cap(samples) < minSize {
+		samples = make([]RefSample, 0, minSize)
```

Comment:
IIUC, we're no longer putting the old `samples` into the pool, not sure if that was intentional.
(Cannot think of any simple way to do that)

Reply:
Previously we would `append` to `samples`, possibly reallocating and discarding the original.
This line skips the first few reallocations but does not change the overall picture.
Over time, the pool tends to contain objects of the largest size needed.

Reply:
Thanks, I just wanted to make sure the case with a monotonically increasing size (where this new approach will allocate a new instead of extending an existing one from the pool (because the pool wouldn't have any)), that case, was considered as rare (which I think it is).